Tasks/WindowsMachineFileCopyV2/RoboCopyJob.ps1

```diff
@@ -231,19 +231,33 @@ param (
         Clean-Target
     }
 
-    $robocopyParameters = Get-RoboCopyParameters -additionalArguments $additionalArguments -fileCopy:$isFileCopy
+    try
+    {
+        $robocopyParameters = Get-RoboCopyParameters -additionalArguments $additionalArguments -fileCopy:$isFileCopy
 
-    $command = "robocopy `"$sourceDirectory`" `"$destinationNetworkPath`" `"$filesToCopy`" $robocopyParameters"                
-    Invoke-Expression $command   
-     
-    if ($LASTEXITCODE -ge 8)
+        $command = "robocopy `"$sourceDirectory`" `"$destinationNetworkPath`" `"$filesToCopy`" $robocopyParameters"
+        Invoke-Expression $command
+        if ($LASTEXITCODE -ge 8)
+        {
+            $errorMessage = Get-VstsLocString -Key "WFC_CopyingFailedConsultRobocopyLogsForMoreDetails"
+            ThrowError -errorMessage $errorMessage -fqdn $fqdn
+        }
+        else
+        {
+            $message = (Get-VstsLocString -Key "WFC_CopyingRecurivelyFrom0to1MachineSucceed" -ArgumentList $sourcePath, $targetPath, $fqdn)
+            Write-Output $message
+        }
+    }
+    finally
     {
-        $errorMessage = Get-VstsLocString -Key "WFC_CopyingFailedConsultRobocopyLogsForMoreDetails"            
-        ThrowError -errorMessage $errorMessage -fqdn $fqdn            
+        if ($machineShare)
+        {
```

Comment:
I see we can regress some of current user what if "Remove-PSDrive" fails

Reply:
this (executing Remove-PSDrive ) already happens earlier in the task when we try to create the destination directory if it is not already there. So, if it should fail here, it should fail earlier as well.